File: Android/FPVR/app/src/main/java/com/solersoft/fpvr/fpvrlib/IGPSInfo.java
package com.solersoft.fpvr.fpvrlib;

/**
 * Provides information from a GPS sensor.
 */
public interface IGPSInfo
{
    //region Public Properties
    /**
     * Gets the altitude in meters.
     */
    public double getAltitude();

    /**
     * Gets the current GPS fix level.
     */
    public GPSFix getGPSFix();

    /**
     * Gets the horizontal Dilution of Precision.
     * @return The hoizontal Dilution of Precision if supported; otherwise 0.
     * For more information see
     * <a href="http://en.wikipedia.org/wiki/Dilution_of_precision_(GPS)">Dilution of Precision</a>.
     */
    public double getHorizontalDilutionOfPrecision();

    /**
     * The latitude in degrees.
     */
    public double getLatitude();

    /**
     * The longitude in degrees.
     */
    public double getLongitude();

    /**
     * Gets the number of satellites visible.
     * @return The number of satellites visible if supported; otherwise 0.
     */
    public int getSatellitesVisible();

    /**
     * Gets the vertical Dilution of Precision.
     * @return The vertical Dilution of Precision if supported; otherwise 0.
     * For more information see
     * <a href="http://en.wikipedia.org/wiki/Dilution_of_precision_(GPS)">Dilution of Precision</a>.
     */
    public double getVerticalDilutionOfPrecision();

    /**
     * The vertical speed in meters per second. This is also known as climb rate.
     * @return The vertical speed in meters per second if supported; otherwise 0.
     */
    public double getVerticleSpeed();
    //endregion
}

File: Lib/FPVR/Interfaces/IGPSInfo.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace FPVR
{
    /// <summary>
    /// Defines the different types of GPS Fix.
    /// </summary>
    public enum GPSFix
    {
        Unknown,
        None,
        Fix2D,
        Fix3D,
        Differential
    };

    /// <summary>
    /// Provides information from a GPS sensor.
    /// </summary>
    public interface IGPSInfo
    {
        /// <summary>
        /// The altitude in meters.
        /// </summary>
        double Altitude { get; }

        /// <summary>
        /// Gets the current GPS fix level.
        /// </summary>
        GPSFix GPSFix { get; }

        /// <summary>
        /// Gets the horizontal Dilution of Precision.
        /// </summary>
        /// <value>
        /// The hoizontal Dilution of Precision if supported; otherwise 0.
        /// </value>
        /// For more information see 
        /// <see href="http://en.wikipedia.org/wiki/Dilution_of_precision_(GPS)">Dilution of Precision</see>.
        double HorizontalDilutionOfPrecision { get; }

        /// <summary>
        /// The latitude in degrees.
        /// </summary>
        double Latitude { get; }
        
        /// <summary>
        /// The longitude in degrees.
        /// </summary>
        double Longitude { get; }

        /// <summary>
        /// Gets the number of satellites visible.
        /// </summary>
        /// <value>
        /// The number of satellites visible if supported; otherwise 0.
        /// </value>
        uint SatellitesVisible { get; }

        /// <summary>
        /// Gets the vertical Dilution of Precision.
        /// </summary>
        /// <value>
        /// The vertical Dilution of Precision if supported; otherwise 0.
        /// </value>
        /// For more information see 
        /// <see href="http://en.wikipedia.org/wiki/Dilution_of_precision_(GPS)">Dilution of Precision</see>.
        double VerticalDilutionOfPrecision { get; }

        /// <summary>
        /// The vertical speed in meters per second.
        /// </summary>
        /// <value>
        /// The vertical speed in meters per second if supported; otherwise 0.
        /// </value>
        /// <remarks>
        /// This is also known as climb rate.
        /// </remarks>
        double VerticleSpeed { get; }
    }
}
